tables/AlternateMans/StokesIStManColumn.h

```diff
@@ -0,0 +1,240 @@
+#ifndef CASACORE_STOKES_I_ST_MAN_COLUMN_H_
+#define CASACORE_STOKES_I_ST_MAN_COLUMN_H_
+
+#include <casacore/tables/DataMan/StManColumn.h>
+
+#include <casacore/casa/Arrays/Array.h>
+#include <casacore/casa/Arrays/IPosition.h>
+
+#include "BufferedColumnarFile.h"
+
+#include <optional>
+
+namespace casacore {
+
+class StokesIStMan;
+
+/**
+ * Expands @p n values from single Stokes I values
+ * to have 4 values, in place. This implies the array
+ * should have place to store n*4 values.
+ */
+template <typename T>
+void ExpandFromStokesI(T *data, size_t n) {
+  for (size_t i = 0; i != n; ++i) {
+    const size_t index = n - i - 1;
+    const T value = data[index];
+    data[index * 4] = value;
+    data[index * 4 + 1] = T();
+    data[index * 4 + 2] = T();
+    data[index * 4 + 3] = value;
+  }
+}
+
+template <typename T>
+inline T *TransformToStokesI(const T *input, char *buffer, size_t n) {
+  for (size_t i = 0; i != n; ++i) {
+    // Placement new is used, because the lifetime of type T needs
+    // to be started.
+    new (&buffer[i * sizeof(T)]) T((input[i * 4] + input[i * 4 + 3]) * T(0.5));
+  }
+  return reinterpret_cast<T *>(buffer);
+}
+
+template <>
+inline bool *TransformToStokesI(const bool *input, char *buffer, size_t n) {
+  for (size_t i = 0; i != n; ++i) {
+    new (&buffer[i]) bool(input[i * 4] || input[i * 4 + 3]);
+  }
+  return reinterpret_cast<bool *>(buffer);
+}
+
+constexpr size_t SizeOfType(DataType dtype) {
```

Comment:
This should exist somewhere else, I hope?

Reply:
It doesn't look like it, I couldn't find it. The way the read/write functions in the other cols seem to work is through overloading, e.g.:
```
    case TpUChar:
        uChar valuc;
        get (rownr, &valuc);
        value = DComplex ((double) valuc);
        return;
    case TpShort:
        Short vals;
        get (rownr, &vals);
        value = DComplex ((double) vals);
        return;
[..]
```
This causes a large number of long `case` statements so I'd say this method is more elegant :). However, this function is more appropriately placed in the `casa/Utilities/DataType.h` file, that way it can be found easier next time it is needed.